OpenRA.Mods.Common/Traits/Render/WithAttackAnimation.cs

```diff
@@ -34,68 +34,90 @@ public class WithAttackAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>,
 		[Desc("Should the animation be delayed relative to preparation or actual attack?")]
 		public readonly AttackDelayType DelayRelativeTo = AttackDelayType.Preparation;
 
-		public object Create(ActorInitializer init) { return new WithAttackAnimation(init, this); }
+		[Desc("Which sprite body to modify.")]
+		public readonly string BodyName = "body";
+
+		public override object Create(ActorInitializer init) { return new WithAttackAnimation(init, this); }
+
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			var matches = ai.TraitInfos<WithSpriteBodyInfo>().Where(w => BodyName == w.Name);
```

Comment:
`.Count(w => BodyName == w.Name)` can be used instead of `.Where(w => BodyName == w.Name)`, removing an unecessary intermediate enumerable.